src/v/cloud_storage/materialized_resources.cc

```diff
@@ -115,84 +105,159 @@ ss::future<> materialized_resources::start() {
     co_return;
 }
 
+inline ssize_t projected_remote_segment_reader_memory_usage() {
```

Comment:
Do we actually need `ssize_t` here?

Reply:
The projected value is compared to the actual estimate and the result of the subtraction can go negative. Saves us some explicit typecasts.